lib/active_shipping/carriers/fedex.rb

```diff
@@ -539,23 +539,28 @@ def parse_transit_times(times)
 
     def extract_address(document, possible_node_names)
       node = nil
+      args = {}
       possible_node_names.each do |name|
         node = document.at(name)
         break if node
       end
 
-      args = if node && node.at('CountryCode')
-        {
-          :country => node.at('CountryCode').text,
-          :province => node.at('StateOrProvinceCode').text,
-          :city => node.at('City').text
-        }
-      else
-        {
-          :country => ActiveUtils::Country.new(:alpha2 => 'ZZ', :name => 'Unknown or Invalid Territory', :alpha3 => 'ZZZ', :numeric => '999'),
-          :province => 'unknown',
-          :city => 'unknown'
-        }
+      if node
+        if node.at('CountryCode')
+          args[:country] = node.at('CountryCode').text
+        else
+          args[:country] = ActiveUtils::Country.new(:alpha2 => 'ZZ', :name => 'Unknown or Invalid Territory', :alpha3 => 'ZZZ', :numeric => '999')
+        end
+        if node.at('StateOrProvinceCode')
+          args[:province] = node.at('StateOrProvinceCode').text
+        else
+          args[:province] = 'unknown'
+        end
+        if node.at('City')
```

Comment:
How would you feel about just:
`args[:city] = node.at('City').try(:text) || 'unknown'`